### PR TITLE
add FileChangeTime and FileLastWrittenMode

### PR DESCRIPTION
**Pull request by microzchang**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-files-shares/src/share_directory_client.cpp [hunk 1]
```diff
@@ -146,6 +146,15 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
...
     }
+    if (options.SmbProperties.ChangedOn.HasValue())
+    {
+      protocolLayerOptions.FileChangeTime = options.SmbProperties.ChangedOn.Value().ToString(
+          Azure::DateTime::DateFormat::Rfc3339, DateTime::TimeFractionFormat::AllDigits);
+    }
+    else
+    {
+      protocolLayerOptions.FileChangeTime = std::string(FileDefaultTimeValue);
+    }
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/src/share_directory_client.cpp:157]:
```suggestion

```

leave the default behavior to storage service

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-shares/src/share_directory_client.cpp:157]: ditto for others

File: sdk/storage/azure-storage-files-shares/src/share_file_client.cpp [hunk 2]
```diff
@@ -346,6 +355,15 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
...
       protocolLayerOptions.FileLastWriteTime = std::string(FileCopySourceTime);
     }
+    if (options.SmbProperties.ChangedOn.HasValue())
+    {
+      protocolLayerOptions.FileChangeTime = options.SmbProperties.ChangedOn.Value().ToString(
+          Azure::DateTime::DateFormat::Rfc3339, DateTime::TimeFractionFormat::AllDigits);
+    }
+    else
+    {
+      protocolLayerOptions.FileChangeTime = std::string(FileCopySourceTime);
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/src/share_file_client.cpp:365]: I think the default behavior is wrong.
According to the [REST API reference](https://docs.microsoft.com/en-us/rest/api/storageservices/copy-file), the default behavior is the time when the copy operation is initiated.

Anyway, let's just remove this `else` clause and let service decide the default behavior,

**Review by Jinming-Hu — CHANGES_REQUESTED** [review]

File: sdk/storage/azure-storage-files-shares/src/share_file_client.cpp (hunk 2 above)
Comment (microzchang) [reply sdk/storage/azure-storage-files-shares/src/share_file_client.cpp:365]: Yeah, I'll fix it.

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**